tracecat/actions/integrations/iam/okta.py

```diff
@@ -28,6 +28,36 @@
 """
 
 
+@registry.register(
+    default_title="Find Okta User",
+    description="Find an Okta user by login username or email",
+    display_group="Okta",
+    namespace="integrations.okta",
+    secrets=[okta_secret],
+)
+async def find_okta_user(
+    username_or_email: Annotated[
+        str,
+        Field(..., description="Login username or e-mail to find"),
+    ],
+) -> list:
+    api_token = os.getenv("OKTA_API_TOKEN")
+    base_url = os.getenv("OKTA_BASE_URL")
+    headers = {
+        "Authorization": f"SSWS {api_token}",
+        "Accept": "application/json",
+        "Content-Type": "application/json",
+    }
+
+    async with httpx.AsyncClient() as client:
+        response = await client.get(
+            f"{base_url}/api/v1/users?search=profile.login%20eq%20%22{username_or_email}%22%20or%20profile.email%20eq%20%22{username_or_email}%22",
+            headers=headers,
+        )
```

Comment:
```suggestion
        response = await client.get(
            f"{base_url}/api/v1/users",
            params={ ... }.
            headers=headers,
        )
```

Reply:
I can, the reason I left it in the uri is that there is only one parameter; it's encoded that way due to how Okta wants the search query passed.  Basically, this one would be pulled out and the ugly search query would be the only item in the param list:

`params={search=f" profile.login%20eq%20%22{username_or_email}%22%20or%20profile.email%20eq%20%22{username_or_email}%22"}`

Not a problem to change though.

Reply:
@mattdurant  Gotcha. (I know this is an outdated thread but just gonna put this here) I think it's easier to read if we format it without the encoding, and if Okta requires it url-encoded, `httpx` does perform the urlsafe percent encoding in the method (unless I'm misunderstanding what you mean specifically by encoding?)

Something like this should work:
```python
params = {
    "search": (
        f'profile.login eq "{username_or_email}" or profile.email eq "{username_or_email}"'
    )
}
response = await client.get(f"{base_url}/api/v1/users", headers=headers, params=params)
return response
```

Am I missing something here?

Reply:
Ah, I was not aware of httpx performing the encoding for you, yes that will work.  I thought in your original comment you wanted it broken out due to being multiple parameters in the string, my mistake.

Reply:
No worries, all good!